src/main/java/com/weaverplatform/import/xmi/XmiImporter.java

```diff
@@ -0,0 +1,270 @@
+package com.weaverplatform.import.xmi;
+
+import com.jcabi.xml.XML;
+import com.jcabi.xml.XMLDocument;
+import com.sun.tools.doclets.internal.toolkit.util.DocFinder;
+import com.weaverplatform.sdk.Entity;
+import com.weaverplatform.sdk.EntityType;
+import com.weaverplatform.sdk.RelationKeys;
+import com.weaverplatform.sdk.Weaver;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.List;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import java.util.UUID;
+
+/**
+ * Created by Jonathan Smit, Sysunite 2016
+ */
+public class XmiImporter {
+
+  private Weaver WEAVER;
+  private boolean USE_UNIX_FILEPATH;
+  private String FILE;
+
+  /**
+   * Constructor
+   * @param weaver instance
+   * @param file specify as filename i.e. "file.xml" or unixpath i.e. "/usr/lib/input.xml"
+   * @param useUnixPath "specify TRUE if @param file equals an unixpath otherwise False (read from classpath test dir)
+   */
+  public XmiImporter(Weaver weaver, String file, boolean useUnixPath){
+    WEAVER = weaver;
+    FILE = file;
+    USE_UNIX_FILEPATH = useUnixPath;
+  }
+
+  /**
+   * Run standalone
+   * @param args
+   * args[0] = weaver connection uri i.e. http://weaver:port
+   * args[1] = file (see also: constructor @param file)
+   */
+  public static void main(String[] args){
+
+    String weaverUrl = args[0];
+    String file = args[1];
+
+    if(weaverUrl != null && file != null) {
+
+      System.out.println(weaverUrl);
+      System.out.println(file);
+
+      Weaver weaver = new Weaver();
+      weaver.connect(weaverUrl);
+
+      XmiImporter xmiImporter = new XmiImporter(weaver, file, true);
+      String xmi = null;
+      try {
+        xmi = IOUtils.toString(xmiImporter.read());
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+      xmi = xmi.replaceAll("UML:", "UML.");
```

Comment:
From here on some XMI processing appears to happen. This functionality should be extracted to its own method, as the main is getting quite long and complex, extracting functionality should make it more clear what is going on. (3 nested for loops make it pretty hard to see at a glance what is going on, this is a code smell).